cdap-watchdog/src/main/java/co/cask/cdap/logging/plugins/FixedWindowRollingPolicy.java

```diff
@@ -0,0 +1,191 @@
+      determineCompressionMode();
+    } else {
+      addError(FNP_NOT_SET);
+      addError(CoreConstants.SEE_FNP_NOT_SET);
+      throw new IllegalStateException(FNP_NOT_SET + CoreConstants.SEE_FNP_NOT_SET);
+    }
+
+    if (maxIndex < minIndex) {
+      addWarn("MaxIndex (" + maxIndex + ") cannot be smaller than MinIndex ("
+                + minIndex + ").");
+      addWarn("Setting maxIndex to equal minIndex.");
+      maxIndex = minIndex;
+    }
+
+    if ((maxIndex - minIndex) > MAX_WINDOW_SIZE) {
+      addWarn("Large window sizes are not allowed.");
+      maxIndex = minIndex + MAX_WINDOW_SIZE;
+      addWarn("MaxIndex reduced to " + maxIndex);
+    }
+
+    IntegerTokenConverter itc = fileNamePattern.getIntegerTokenConverter();
+
+    if (itc == null) {
+      throw new IllegalStateException("FileNamePattern ["
+                                        + fileNamePattern.getPattern()
+                                        + "] does not contain a valid IntegerToken");
+    }
+
+    if (compressionMode == CompressionMode.ZIP) {
+      String zipEntryFileNamePatternStr = transformFileNamePatternFromInt2Date(fileNamePatternStr);
+      zipEntryFileNamePattern = new FileNamePattern(zipEntryFileNamePatternStr, context);
+    }
+
+    super.start();
+  }
+
+  private String transformFileNamePatternFromInt2Date(String fileNamePatternStr) {
+    String slashified = FileFilterUtil.slashify(fileNamePatternStr);
+    String stemOfFileNamePattern = FileFilterUtil.afterLastSlash(slashified);
+    return stemOfFileNamePattern.replace("%i", "%d{" + ZIP_ENTRY_DATE_PATTERN + "}");
+  }
+
+  /**
+   * Given the FileNamePattern string, this method determines the compression
+   * mode depending on last letters of the fileNamePatternStr. Patterns ending
+   * with .gz imply GZIP compression, endings with '.zip' imply ZIP compression.
+   * Otherwise and by default, there is no compression.
+   */
+  @Override
+  protected void determineCompressionMode() {
+    if (fileNamePatternStr.endsWith(".gz")) {
+      addInfo("Will use gz compression");
+      compressionMode = CompressionMode.GZ;
+    } else if (fileNamePatternStr.endsWith(".zip")) {
+      addInfo("Will use zip compression");
+      compressionMode = CompressionMode.ZIP;
+    } else {
+      addInfo("No compression will be used");
+      compressionMode = CompressionMode.NONE;
+    }
+  }
+
+  @Override
+  public void rollover() throws RolloverFailure {
+
+    // Inside this method it is guaranteed that the hereto active log file is
+    // closed.
+    // If maxIndex <= 0, then there is no file renaming to be done.
+    if (maxIndex >= 0) {
+      try {
+        // Delete the oldest file, to keep Windows happy.
+        String fileName = fileNamePattern.convertInt(maxIndex);
+        Location deleteLocation = activeParentLocation.append(fileName);
+        if (deleteLocation.exists()) {
+          deleteLocation.delete();
+        }
```

Comment:
That is part of test right, At a time we should only have maxIndex number of files when we have many log events